### PR TITLE
[flant-integration] Add cpu and memory metrics for node groups

### PR DESCRIPTION
**Pull request by vitaliy-sn**
## Description

- Added metrics for the number of cpu and memory in node groups.
- Added collection of metrics for control-plane components.

## Checks

image
image

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: flant-integration
type: feature
summary: Added metrics for the number of cpu and memory in node groups.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/static'...

**Review by z9r5 — APPROVED** [review]

File: ee/be/modules/600-flant-integration/images/flant-pricing/hooks/node_group_resources_metrics.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      group: ngs
+      keepFullObjectsInMemory: false
+      queue: /node_group_resources_metrics
+      apiVersion: deckhouse.io/v1
+      kind: NodeGroup
+      jqFilter: |
+        {
+          "name": .metadata.name,
+          "labels": .spec.labels,
+          "taints": .spec.taints
```
Comment (nabokihms) [inline ee/be/modules/600-flant-integration/images/flant-pricing/hooks/node_group_resources_metrics.sh:24]: I believe that `.spec.taints` and `.spec.labels` is a mistake. It should be at lease `.spec.nodeTemplate`.

File: ee/be/modules/600-flant-integration/images/flant-pricing/hooks/node_group_resources_metrics.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  fi
+
+  for node_group_name in $(context::jq -cr '.snapshots.ngs[].filterResult.name'); do
+    ng_capacity="$(jq -cr --arg ng_name "$node_group_name" '.[$ng_name] // ""' <<< "$ngs_capacity")"
+    if [ "$ng_capacity" == "" ]; then
+      continue
+    fi
+    cpu="$(jq -cr '.CPU' <<< "$ng_capacity")"
+    memory="$(jq -cr '.memory' <<< "$ng_capacity")"
+    labels="$(context::jq -cr --arg ng_name "$node_group_name" '.snapshots.ngs[].filterResult | select(.name == $ng_name) | ((.taints // [] | .[] | {("taint-"+.effect): .key}) // {}) as $t | (.labels // {}) as $l | $t * $l * {"name": .name}')"
```
Comment (nabokihms) [inline ee/be/modules/600-flant-integration/images/flant-pricing/hooks/node_group_resources_metrics.sh:50]: Do we need to export all labels? For what reason?

**Review by nabokihms — CHANGES_REQUESTED** [review]

File: ee/be/modules/600-flant-integration/hooks/pricing/envs_from_all_nodes.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+func ApplyNodeCapacityFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	node := &v1.Node{}
+	err := sdk.FromUnstructured(obj, node)
+	if err != nil {
+		return nil, err
+	}
+
+	n := &NodeCapacity{}
+
+	if _, ok := node.ObjectMeta.Labels["node.deckhouse.io/group"]; !ok {
```
Comment (name212) [inline ee/be/modules/600-flant-integration/hooks/pricing/envs_from_all_nodes.go:40]: ObjectMeta is unnecessary

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Review by asviel — APPROVED** [review]

File: ee/be/modules/600-flant-integration/templates/pricing/daemonset.yaml [hunk 4]
```diff
@@ -117,6 +117,8 @@ spec:
           value: {{ .Values.flantIntegration.kubeall.kubeconfig | quote }}
         - name: FP_KUBEALL_CONTEXT
           value: {{ .Values.flantIntegration.kubeall.context | quote }}
+        - name: FP_NODE_GROUPS_CAPACITY
+          value: {{ .Values.flantIntegration.internal.nodeGroupsCapacity | toJson | quote }}
```
[the remark this answers is not in the document]
Comment (vitaliy-sn) [reply ee/be/modules/600-flant-integration/templates/pricing/daemonset.yaml]: This is no longer relevant. The flant-pricing hook was rewritten in python and the go hook was no longer needed.

File: ee/be/modules/600-flant-integration/images/flant-pricing/hooks/node_group_resources_metrics.sh (hunk 1 above)
Comment (vitaliy-sn) [reply ee/be/modules/600-flant-integration/images/flant-pricing/hooks/node_group_resources_metrics.sh]: Yes, this is a mistake.

File: ee/be/modules/600-flant-integration/images/flant-pricing/hooks/node_group_resources_metrics.sh (hunk 2 above)
Comment (vitaliy-sn) [reply ee/be/modules/600-flant-integration/images/flant-pricing/hooks/node_group_resources_metrics.sh]: Currently we only export the following labels:

```python
        labels = {
            "is_master": is_master,
            "is_system": is_system,
            "is_monitoring": is_monitoring,
            "is_frontend": is_frontend,
        }
```

**Review by RomanenkoDenys — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/aws'...

**Review by RomanenkoDenys — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [8202](https://github.com/deckhouse/deckhouse/pull/8202) to the branch [release-1.59](https://github.com/deckhouse/deckhouse/tree/release-1.59) successful!

**Pull request merged**